how-to/customize-workspace/client/src/modules/composite/windows/actions.ts

```diff
@@ -0,0 +1,98 @@
+import type OpenFin from "@openfin/core";
+import type {
+	CustomActionPayload,
+	CustomActionsMap,
+	WorkspacePlatformModule
+} from "@openfin/workspace-platform";
+import type { ActionHelpers, Actions } from "customize-workspace/shapes/actions-shapes";
+import type { Logger, LoggerCreator } from "customize-workspace/shapes/logger-shapes";
+import type { ModuleDefinition } from "customize-workspace/shapes/module-shapes";
+import { getAllVisibleWindows } from "./helper";
+
+/**
+ * Implement the actions.
+ */
+export class WindowActions implements Actions {
+	/**
+	 * The helper methods to use.
+	 */
+	private _helpers: ActionHelpers;
+
+	/**
+	 * The helper methods to use.
+	 */
+	private _logger: Logger;
+
+	/**
+	 * Initialize the module.
+	 * @param definition The definition of the module from configuration include custom options.
+	 * @param createLogger For logging entries.
+	 * @param helpers Helper methods for the module to interact with the application core.
+	 * @returns Nothing.
+	 */
+	public async initialize(
+		definition: ModuleDefinition,
+		createLogger: LoggerCreator,
+		helpers: ActionHelpers
+	): Promise<void> {
+		this._logger = createLogger("WindowActions");
+		this._helpers = helpers;
+	}
+
+	/**
+	 * Get the actions from the module.
+	 */
+	public async get(platform: WorkspacePlatformModule): Promise<CustomActionsMap> {
+		const actionMap: CustomActionsMap = {};
+
+		actionMap["window-showall"] = async (payload: CustomActionPayload) => {
```

Comment:
Can we add an extra hyphen in the names, i.e. `show-all`, `hide-all`, `hide-others`

Reply:
done